ament_cppcheck/ament_cppcheck/main.py

```diff
@@ -170,7 +170,12 @@ def main(argv=sys.argv[1:]):
             'severity': error.get('severity'),
             'msg': error.get('verbose'),
         }
-        report[filename].append(data)
+        if os.path.relpath(os.path.realpath(filename)) in report:
+            filename = os.path.relpath(os.path.realpath(filename))
+        # in the case where relative and absolute paths are mixed for paths and
+        # include_dirs cppcheck might return duplicate results
+        if not any(data == d for d in report[filename]):
```

Comment:
nit, [same as](https://docs.python.org/3/reference/expressions.html#membership-test-operations)

```python3
if data not in report[filename]:
```

Reply:
I don't think that is the case. `in` will compare the dictionaries based on their id where as the `==` operator compares them based on their content.

Reply:
> in will compare the dictionaries based on their id where as the == operator compares them based on their content.


[`in` uses `==` too](https://docs.python.org/3/reference/expressions.html#membership-test-operations)

> the expression `x in y` is equivalent to `any(x is e or x == e for e in y)`

Reply:
Hm, I am sure I tried it before switching to `not any(...)` and it didn't work. I changed it to `not in` in 6434b6502ed5a9639f3d9839417275b5c2acd89b and it works as you mentioned.

Thanks for pointing this out - much easier to read this way.